src/main/scala/com/kinja/amqp/MessageBufferProcessor.scala

```diff
@@ -92,10 +92,11 @@ class MessageBufferProcessor(
 		val batchSizeWithExtraGap = batchSize * 2
 		val messages = messageStore.loadLockedMessages(batchSizeWithExtraGap)
 		scala.util.Random.shuffle(producers).foreach {
-			case (exchange, producer) =>
-				val messagesToProducer = messages.filter(_.exchangeName == producer.exchange.name)
+			case (exchange: String, producer: AmqpProducerInterface) =>
```

Comment:
Ok, not `isInstanceOf` but `asInstanceOf`.

Given this source:

```scala
object Test {

	val pair: (String, Int) = ("a", 1)

	val x = pair match {
		case (s: String, i: Int) => "case1"
		case _ => "case2"
	}
}
```

The Scala compiler generates this code (after `erasure` phase):

```scala
package <empty> {
  object Test extends Object {
    def <init>(): Test.type = {
      Test.super.<init>();
      ()
    };
    private[this] val pair: Tuple2 = new Tuple2("a", scala.Int.box(1));
    <stable> <accessor> def pair(): Tuple2 = Test.this.pair;
    private[this] val x: String = {
      case <synthetic> val x1: Tuple2 = Test.this.pair();
      case6(){
        if (x1.ne(null))
          {
            val s: String = x1._1().$asInstanceOf[String]();
            if (s.ne(null))
              matchEnd5("case1")
            else
              case7()
          }
        else
          case7()
      };
      case7(){
        matchEnd5("case2")
      };
      matchEnd5(x: String){
        x
      }
    };
    <stable> <accessor> def x(): String = Test.this.x
  }
}
```

While without the explicit type ascriptions it looks like this:
```scala
package <empty> {
  object Test extends Object {
    def <init>(): Test.type = {
      Test.super.<init>();
      ()
    };
    private[this] val pair: Tuple2 = new Tuple2("a", scala.Int.box(1));
    <stable> <accessor> def pair(): Tuple2 = Test.this.pair;
    private[this] val x: String = {
      case <synthetic> val x1: Tuple2 = Test.this.pair();
      case4(){
        if (x1.ne(null))
          matchEnd3("case1")
        else
          case5()
      };
      case5(){
        matchEnd3("case2")
      };
      matchEnd3(x: String){
        x
      }
    };
    <stable> <accessor> def x(): String = Test.this.x
  }
}
```